streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -476,7 +476,9 @@ class BatchedWriteAheadLogSuite extends CommonWriteAheadLogTests(
     val batchedWal = new BatchedWriteAheadLog(wal, sparkConf)
 
     val e = intercept[SparkException] {
-      val buffer = mock[ByteBuffer]
+      // SPARK-40731: Use a 0 size `ByteBuffer` instead of `mock[ByteBuffer]`
+      // due to mockito 4 can't mock/spy sealed class
+      val buffer = ByteBuffer.allocate(0)
```

Comment:
or should we 

```
      val buffer = if (SystemUtils.isJavaVersionAtMost(JavaVersion.JAVA_17)) {
        mock[ByteBuffer]
      } else {
        ByteBuffer.allocate(0)
      }
```

but this need add commons-lang3 as test dependency for `steaming` module

Reply:
[012c719](https://github.com/apache/spark/pull/41815/commits/012c719518e321e7c7e0ddc965f0592519229c43) change to this way